### PR TITLE
Make MModeTransform slice mmodes at tel_mmax if requested

### PR DESCRIPTION
**Pull request by cahofer**
I would like to have the capability to slice the mmodes at the telescope mmax if requested. 
(when we have more time samples in the visibilities than 2*mmax + 1)

### CODE REVIEW

File: draco/analysis/transform.py [hunk 1]
```diff
@@ -432,7 +432,18 @@ class MModeTransform(task.SingleTask):
...
+
+    Attributes
+    ----------
+    set_telmmax : bool
+        Whether to slice the data at the maximum m the telescope can resolve
+        after fourier transforming. Default : false.
     """
+    set_telmmax = config.Property(proptype=bool, default=False)
+
+    def setup(self, manager):
```
Comment (jrs65) [inline draco/analysis/transform.py]: This introduces a mandatory dependency on the `manager` but that's actually unnecessary if you don't set `set_telmmax`.

I would change this to make manager optional `i.e. manager=None`, and then if `manager` is not None this is equivalent to setting `set_telmmax`. Then I'd remove the `set_telmmax` option entirely.

File: draco/analysis/transform.py [hunk 2]
```diff
@@ -453,8 +464,13 @@ def process(self, sstream):
         # variance for the m-modes
         weight_sum = sstream.weight[:].sum(axis=-1)
 
+        if self.set_telmmax is True:
```
Comment (jrs65) [inline draco/analysis/transform.py]: With that I said above this clause would just change to `self.telescope is not None`

File: draco/analysis/transform.py (hunk 1 above)
Comment (cahofer) [reply draco/analysis/transform.py]: Ok that makes sense. I made the changes and just going to quickly test the code.

**Comment by cahofer** [comment]
Let me know if the changes I implemented were the ones you had in mind. Now the manager is optional.

File: draco/analysis/transform.py [hunk 3]
```diff
@@ -433,6 +433,12 @@ class MModeTransform(task.SingleTask):
 
     Currently ignores any noise weighting.
     """
+    def setup(self, manager=None):
```
Comment (jrs65) [inline draco/analysis/transform.py]: The docstring for this function should explain the behaviour here.

File: draco/analysis/transform.py [hunk 4]
```diff
@@ -453,8 +459,13 @@ def process(self, sstream):
         # variance for the m-modes
         weight_sum = sstream.weight[:].sum(axis=-1)
 
+        if self.telescope is not None:
+            setmmax = self.telescope.mmax
```
Comment (jrs65) [inline draco/analysis/transform.py]:
```suggestion
            mmax = self.telescope.mmax
```

File: draco/analysis/transform.py [hunk 5]
```diff
@@ -453,8 +459,13 @@ def process(self, sstream):
...
+        else:
+            setmmax = None
```
Comment (jrs65) [inline draco/analysis/transform.py]:
```suggestion
            mmax = None
```

File: draco/analysis/transform.py [hunk 6]
```diff
@@ -453,8 +459,13 @@ def process(self, sstream):
...
+
         # Construct the array of m-modes
-        marray = _make_marray(sstream.vis[:])
+        marray = _make_marray(sstream.vis[:], setmmax)
```
Comment (jrs65) [inline draco/analysis/transform.py]:
```suggestion
        marray = _make_marray(sstream.vis[:], mmax)
```

File: draco/analysis/transform.py (hunk 4 above)
Comment (jrs65) [inline draco/analysis/transform.py]: This variable name doesn't make sense. `setxyz` sounds more like a binary parameter that sets some behaviour, whereas it actually represents the `mmax` itself.

File: draco/analysis/transform.py (hunk 3 above)
Comment (jrs65) [reply draco/analysis/transform.py]: You might also want to explain the behaviour in the class docstring. Only one location needs a full account though.

**Comment by cahofer** [comment]
Okay, I added a docstring for setup and changed the cosmetic.

**Pull request merged**